enhancements/installer/kubernete-slb-owned-api-loadbalancers.md

```diff
@@ -0,0 +1,508 @@
+
+- The controllers create cloud resources based on the [UID of the service object](https://github.com/kubernetes/cloud-provider/blob/master/cloud.go#L88). 
+    This is deprecated method and recommends the cloud provider implementations use more useful names. There are no cloud providers that have moved to a 
+    better naming scheme and there are no enhancements for such a change. The installer can only create the resources that are correctly adopted if the names 
+    are predictable. And even if the installer could create these resources, there would be very high tight coupling of the implementation of controllers and 
+    the installer.
+- The bootstrap host needs to be part of the backend of the load balancers, but the bootstrap host is not a node in the cluster. Therefore, the controllers 
+    when adopting the resources will remove the bootstrap host from the backend before the control-plane endpoints were running making the bootstrapping more
+    fragile.
+
+Using DNS allows the installer to abstract the bootstrap host and load balancer behind the same DNS and k8s controllers can own the lifecycle of SLB from 
+creation to mangement. This keeps the installer's bootstrapping workflow simpler.
+
+#### Using installer created static ips
+
+The installer will create static ips for the loadbalancers created by the cluster for API and API-INT. This allows the installer to pre-seed the DNS records 
+with ips of the `bootstrap` machine and these ips and also update the records after bootstrapping is complete to include only the loadbalancer ips.
+
+An alternative would have been to allow the cluster to use dynamic ips and then update the records itself, but there are several caveats like,
+
+- Accessing the cloud DNS APIs from the cluster has proven to be difficult esp. on AWS where in disconnected(air-gapped) environment the cluster 
+    cannot communicate to the route53 API because of lack of VPC endpoints.
+- Allowing the cluster to perform the DNS actions for such a crucial endpoint means we will be forced to support various DNS providers. Restricting 
+    the DNS operations to installer allows the in-cluster manager to be simple for the time being as we understand the landscape a little bit more.
+
+#### RRDNS and unusuable ips
+
+During bootstrapping where the installer creates the DNS records with both the `bootstrap` host IPs and static IPs, the clients like the kubelet, 
+network operator can try to connect to the static IP which is not yet routable to anything. Also, when the bootstrapping is complete and the 
+bootstrap-control-plane has itself shutdown on the `bootstrap` host, again clients previously can still end up trying to connect to the bootstrap ip 
+which is no longer serving traffic.
+
+In both these situations the clients are going to see some errors for the time being, but in both these situations RRDNS should allow the clients to 
+find and use the correct ip addresses for the API.
```

Comment:
we have quite some kubectl scripting going on, e.g. for csr approval. These scripts will have to be prepared to handle errors. Today they don't.

Also do we know the DNS resolution behaviour of client-go and Go in general? When does it relookup records in order to get another IP (a good one after error) ?